tests/python_client/testcases/test_search.py

```diff
@@ -6660,3 +6687,540 @@ def test_range_search_with_consistency_session(self, nq, dim, auto_id, _async):
+                                  1000, default_search_exp, partition_names=[partition_w1.name])[0]
+        assert len(res[0]) == 200
+        collection_w.search(vectors[:1], default_search_field, default_search_params,
+                            default_limit, default_search_exp, partition_names=[partition_w2.name],
+                            check_task=CheckTasks.err_res,
+                            check_items={ct.err_code: 1, ct.err_msg: 'not loaded'})
+
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_compact_load_partition_release_collection(self):
+        """
+        target: test compact load partition release collection
+        method: 1. create a collection and 2 partitions
+                2. insert data multi times
+                3. compact
+                4. load one partition
+                5. release the collection
+                6. search on the partition -> raise error
+        expected: No exception
+        """
+        collection_w = self.init_collection_general(prefix, partition_num=1)[0]
+        partition_w1, partition_w2 = collection_w.partitions
+        df = cf.gen_default_dataframe_data()
+        # insert data
+        partition_w1.insert(df[:100])
+        partition_w1.insert(df[100:200])
+        partition_w2.insert(df[200:300])
+        # compact
+        collection_w.compact()
+        collection_w.get_compaction_state()
+        # load && release
+        partition_w2.load()
+        collection_w.release()
+        # search
+        collection_w.search(vectors[:1], default_search_field, default_search_params,
+                            default_limit, default_search_exp, partition_names=[partition_w2.name],
+                            check_task=CheckTasks.err_res,
+                            check_items={ct.err_code: 1, ct.err_msg: 'not loaded'})
+
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_load_collection_compact_drop_partition(self):
+        """
+        target: test load collection compact drop partition
+        method: 1. create a collection and 2 partitions
+                2. insert data multi times
+                3. load the collection
+                4. compact
+                5. release one partition and drop
+                6. search on the dropped partition -> raise exception
+                7. search on the non-dropped partition
+        expected: No exception
+        """
+        collection_w = self.init_collection_general(prefix, partition_num=1)[0]
+        partition_w1, partition_w2 = collection_w.partitions
+        df = cf.gen_default_dataframe_data()
+        # insert data
+        partition_w1.insert(df[:100])
+        partition_w1.insert(df[100:200])
+        partition_w2.insert(df[200:300])
+        # load
+        collection_w.load()
+        # compact
+        collection_w.compact()
+        collection_w.get_compaction_state()
+        # release
+        partition_w2.release()
+        partition_w2.drop()
```

Comment:
i think it worth a try of dropping the partition directly without release

Reply:
> i think it worth a try of dropping the partition directly without release

According to issue #23620, if take away this row ‘partition_w2.release()’, an error will be raised.